flink-optimizer/src/main/java/org/apache/flink/optimizer/dag/ReduceNode.java

```diff
@@ -45,10 +46,28 @@ public ReduceNode(ReduceOperatorBase<?, ?> operator) {
 			// case of a key-less reducer. force a parallelism of 1
 			setParallelism(1);
 		}
-		
-		OperatorDescriptorSingle props = this.keys == null ?
-			new AllReduceProperties() :
-			new ReduceProperties(this.keys, operator.getCustomPartitioner());
+
+		OperatorDescriptorSingle props;
+
+		if (this.keys == null) {
+			props = new AllReduceProperties();
+		} else {
+			DriverStrategy combinerStrategy;
+			switch(operator.getCombineHint()) {
+				case OPTIMIZER_CHOOSES:
+					combinerStrategy = DriverStrategy.SORTED_PARTIAL_REDUCE;
```

Comment:
I included the `OPTIMIZER_CHOOSES` case because `JoinHint` and `CrossHint` also have this case. I guess it is there because we might modify what the default is later.